File: Android_Domaci2/app/src/main/java/com/example/domaci2/LoginActivity.java
package com.example.domaci2;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {


    EditText username;
    EditText password;
    Button loginBtn;
    public static final String REG_NAME_KEY = "usernameKey";
//    public static final String REG_PASS_KEY = "passwordKey";


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        init();
    }

    private void init(){
        initview();
        initListeners();
    }

    private void initview(){
        username = findViewById(R.id.logUsername);
        password = findViewById(R.id.logPass);
        loginBtn = findViewById(R.id.loginBtn);
    }
    private void initListeners(){

        SharedPreferences sharedPreferences = getSharedPreferences(getPackageName(), Context.MODE_PRIVATE);
//        String usernameStored = sharedPreferences.getString(REG_NAME_KEY, "");
//        String passwordStored = sharedPreferences.getString(REG_PASS_KEY, "");


        loginBtn.setOnClickListener(v ->{
            String userInput = username.getText().toString();
            String passInput = password.getText().toString();

            System.out.println( userInput + " - " + passInput);

            if (!userInput.equals("") && !passInput.equals("")){
                sharedPreferences
                    .edit()
                    .putString(REG_NAME_KEY, userInput)
//                        .putString(REG_PASS_KEY, password.getText().toString().trim())
                    .apply();


            Intent intent = new Intent(this, ScreenActivity.class);
            intent.putExtra(REG_NAME_KEY, userInput);
            startActivity(intent);
            finish();



            }
            else {
                Toast.makeText(this, "Empty login parameter.", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: Android_Projekat3/app/src/main/java/com/example/projekat3/presentation/view/activities/LoginActivity.kt
package com.example.projekat3.presentation.view.activities

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.projekat3.R

class LoginActivity : AppCompatActivity() {

    private lateinit var loginButton: Button
    private lateinit var passwordInput: EditText
    private lateinit var registerButton: Button
    private lateinit var usernameInput: EditText
    private lateinit var emailInput: EditText


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        initView()
        initListeners()
    }

    private fun initView() {
        registerButton = findViewById(R.id.registerBtn)
        emailInput = findViewById(R.id.emailIn)
        loginButton = findViewById(R.id.loginBtn)
        usernameInput = findViewById(R.id.usernameIn)
        passwordInput = findViewById(R.id.passwordIn)
    }

    /*
    mode je da bi se znalo da li da se stavi novi user na bazu ili da se ucita stari
    ako se uloguje kao ne postojeci user puca program !!!
     */

    private fun initListeners() {
        loginButton.setOnClickListener {
            saveInfo("LOGIN")
        }

        registerButton.setOnClickListener{
            saveInfo("REGISTER")
        }
    }

    private fun saveInfo(mode: String){
        val sharedPreferences = getSharedPreferences(packageName, MODE_PRIVATE)

        val email = emailInput.text.toString()
        val username = usernameInput.text.toString()
        val password = passwordInput.text.toString()

        if (username != "" && password != "" && email != "" && password.length >= 5) {
            sharedPreferences
                .edit()
                .putString("mode", mode)
                .putString("username", username)
                .putString("email", email)
                .putString("password", password)
                .apply()

            val intent = Intent(this, AppActivity::class.java)
            startActivity(intent)
            finish()
        } else {
            Toast.makeText(this, "Error logging in, pass > 5", Toast.LENGTH_SHORT).show()
        }
    }
}